units/migrate_volume_job.go

```diff
@@ -0,0 +1,298 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const (
+	volumeMigrationJobName = "volume-migration"
+)
+
+func init() {
+	registry.AddJobType(volumeMigrationJobName,
+		func() amboy.Job { return makeVolumeMigrationJob() })
+}
+
+type volumeMigrationJob struct {
+	job.Base      `bson:"job_base" json:"job_base" yaml:"job_base"`
+	VolumeID      string             `bson:"volume_id" yaml:"volume_id"`
+	ModifyOptions cloud.SpawnOptions `bson:"modify_options" json:"modify_options" yaml:"modify_options"`
+
+	VolumeDetached        bool   `bson:"volume_detached"`
+	InitialHostID         string `bson:"initial_host_id"`
+	SpawnhostStopJobID    string `bson:"spawnhost_stop_job_id"`
+	InitialHostStopped    bool   `bson:"initial_host_stopped"`
+	InitialHostTerminated bool   `bson:"initial_host_terminated"`
+	NewHostID             string `bson:"new_host_id"`
+
+	env         evergreen.Environment
+	volume      *host.Volume
+	initialHost *host.Host
+}
+
+func makeVolumeMigrationJob() *volumeMigrationJob {
+	j := &volumeMigrationJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    volumeMigrationJobName,
+				Version: 0,
+			},
+		},
+	}
+	return j
+}
+
+func NewVolumeMigrationJob(env evergreen.Environment, volumeID string, modifyOptions cloud.SpawnOptions, ts string) amboy.Job {
+	j := makeVolumeMigrationJob()
+	j.env = env
+	j.SetID(fmt.Sprintf("%s.%s.%s", volumeMigrationJobName, volumeID, ts))
+	j.SetScopes([]string{fmt.Sprintf("%s.%s", volumeMigrationJobName, volumeID)})
+	j.SetEnqueueAllScopes(true)
+	j.VolumeID = volumeID
+	j.ModifyOptions = modifyOptions
```

Comment:
Where do these ModifyOptions come from? The new or old host? I think this is a lot to store; it'd be preferable if we could just look it up when setting up the job. (Although I'm just thinking of convention; I'm not sure if there's a reason or it's just happenstance that we don't usually store much in the job @Kimchelly )

Reply:
These ModifyOptions are the options that a user has configured via a modal for their new host. I'm adding a GraphQL mutation in [EVG-17486](https://jira.mongodb.org/browse/EVG-17486) that will have a similar workflow to the [SpawnHost](https://github.com/evergreen-ci/evergreen/blob/a10726bcd4549772aa18985e2d43cb27d90b87d2/graphql/mutation_resolver.go#L618) mutation; the new volume migration mutation will accept an [APIHost](https://github.com/evergreen-ci/evergreen/blob/f0573cc00ff1af555ca4c7a2d342239ac062b3d6/rest/model/host.go#L15-L38) struct that is converted to the SpawnOptions type.

Since the job is kicked off from a mutation that the user calls I'm not sure if there's anywhere that makes sense to store this information, but let me know if I'm missing anything!

Reply:
I don't think there's an issue with storing large-ish options structs in the Amboy job. This is a normal way to set up an Amboy job when all you have are options to configure the new resource.